io/genwqe/genwqetest.py

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2017 IBM
+# Author: Narasimhan V <sim@linux.vnet.ibm.com>
+
+"""
+This Suite tests the GenWQE Accelarator.
+"""
+
+import os
+import shutil
+import time
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import download
+from avocado.utils.software_manager import SoftwareManager
+
+
+class GenWQETest(Test):
+
+    """
+    Test Class for GenWQE Tests.
+
+    :param device: Name of the GenWQE device
+    :param test_tar_url: URL for the test tarball
+    """
+
+    def setUp(self):
+        """
+        Install genwqe packages, and downloads test tarball.
+        """
+        self.card = self.params.get('device', default='0')
+        url = "http://corpus.canterbury.ac.nz/resources/cantrbry.tar.gz"
+        self.url = self.params.get('test_tar_url', default=url)
+        if not os.path.isdir("/sys/class/genwqe/genwqe%s_card/" % self.card):
+            self.skip("Device %s does not exist" % self.card)
+        smm = SoftwareManager()
+        pkgs = ['genwqe-tools', 'genwqe-zlib']
+        for pkg in pkgs:
+            if not smm.check_installed(pkg) and not smm.install(pkg):
+                self.skip('pkg is needed for the test to be run')
```

Comment:
can you please display package name as well .it help user which package is required ?